main.tf

```diff
@@ -78,6 +80,64 @@ resource "aws_cloudwatch_event_bus" "this" {
     }
   }
 
+  dynamic "log_config" {
+    for_each = var.logging != null ? [var.logging] : []
+    content {
+      include_detail = log_config.value.include_detail
+      level          = log_config.value.level
+    }
+  }
+
+  tags = var.tags
+}
+
+resource "aws_cloudwatch_log_delivery_source" "this" {
+  count = local.create_logging ? 1 : 0
+
+  region = var.region
+
+  name         = coalesce(var.log_delivery_source_name, var.bus_name)
+  log_type     = "${upper(var.logging.level)}_LOGS"
+  resource_arn = aws_cloudwatch_event_bus.this[0].arn
+
+  tags = var.tags
+}
+
+resource "aws_cloudwatch_log_delivery_destination" "this" {
+  for_each = { for k, v in var.logging : k => v if(local.create_logging && contains(["s3", "cloudwatch_logs", "firehose"], k) && try(v.enabled, true) && v != null) }
```

Comment:
this should be simpler once the `log_config` values have been removed from the `log_delivery` variable

also, we haven't validated the `complete` example because it will break (and most of the other examples). You cannot iterate with a ` { for k, v ...}` over a `null` value. So either the `logging` variable default needs to change, or this logic